src/modules/credentials/handlers/OfferCredentialHandler.ts

```diff
@@ -1,17 +1,64 @@
+import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { Handler, HandlerInboundMessage } from '../../../agent/Handler'
+import type { CredentialRecord } from '../repository/CredentialRecord'
 import type { CredentialService } from '../services'
 
+import { createOutboundMessage } from '../../../agent/helpers'
+import { AutoAcceptCredential } from '../../../types'
+import { CredentialUtils } from '../CredentialUtils'
 import { OfferCredentialMessage } from '../messages'
 
 export class OfferCredentialHandler implements Handler {
   private credentialService: CredentialService
+  private agentConfig: AgentConfig
   public supportedMessages = [OfferCredentialMessage]
 
-  public constructor(credentialService: CredentialService) {
+  public constructor(credentialService: CredentialService, agentConfig: AgentConfig) {
     this.credentialService = credentialService
+    this.agentConfig = agentConfig
   }
 
   public async handle(messageContext: HandlerInboundMessage<OfferCredentialHandler>) {
-    await this.credentialService.processOffer(messageContext)
+    const credentialRecord = await this.credentialService.processOffer(messageContext)
+
+    const autoAccept = CredentialUtils.composeAutoAccept(
+      credentialRecord.autoAcceptCredential,
+      this.agentConfig.autoAcceptCredentials
+    )
+
+    if (autoAccept === AutoAcceptCredential.always) {
+      return await this.nextStep(credentialRecord, messageContext)
+    } else if (autoAccept === AutoAcceptCredential.contentApproved) {
+      if (credentialRecord.proposalMessage && credentialRecord.offerMessage) {
+        const proposalValues = CredentialUtils.convertAttributesToValues(
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
shouldn't use `!`. We can assume we have the message, but if we change something in the future this can break